pageserver/src/tenant/storage_layer/layer.rs

```diff
@@ -1915,6 +1915,18 @@ impl ResidentLayer {
             Image(_) => Err(anyhow::anyhow!("image layer")),
         }
     }
+
+    #[cfg(test)]
+    pub(crate) async fn as_image(
+        &self,
+        ctx: &RequestContext,
+    ) -> anyhow::Result<&image_layer::ImageLayerInner> {
+        use LayerKind::*;
+        match self.downloaded.get(&self.owner.0, ctx).await? {
+            Image(ref d) => Ok(d),
+            Delta(_) => Err(anyhow::anyhow!("delta layer")),
+        }
+    }
```

Comment:
I think we need to rename it to `download_as_delta` instead. The function downloads the layer and then converts it into image/delta layer. In this case, I think it would be better to keep anyhow::Result here to avoid having a `Result<Option<X>>` thing.

Reply:
resolved in https://github.com/neondatabase/neon/pull/8006/commits/ea2dfa7a96a5ca9bcb27c459f07c45ac43253be5